pkg/controllers/apply_controller.go

```diff
@@ -202,24 +201,27 @@ func (r *ApplyWorkReconciler) applyUnstructured(
 		return nil, false, err
 	}
 
-	if !findOwnerReference(curObj.GetOwnerReferences(), workObj.GetOwnerReferences()[0]) {
+	if !hasSharedOwnerReference(curObj.GetOwnerReferences(), workObj.GetOwnerReferences()[0]) {
 		// TODO: Block All Owner reference in the Work Manifest.
 		err = fmt.Errorf("this object is not owned by the work-api")
 		klog.V(5).InfoS("This object is not owned by the work-api.", "gvr", gvr, "obj", workObj.GetName(), "err", err)
 		return nil, false, err
 	}
 
-	// Compare and update the unstrcuctured.
-	needUpdate := false
-	if !isSameUnstructuredMeta(workObj, curObj) {
+	// Compare the unstructured object and update if needed.
+	updateWarranted := isUpdateWarranted(workObj, curObj)
```

Comment:
when is the annotation set on the workObj?

Reply:
At first I thought I would need to calculate it and then compare to the annotation set on the existing resource in the member cluster. But we set it at line ~186 when we first enter the applyUnstructured func. The code flow guarantees it was set before the compare occurs.